File: sdks/java/basys.sdk/src/main/java/org/eclipse/basyx/aas/impl/metamodel/hashmap/aas/qualifier/haskind/HasKind.java
package org.eclipse.basyx.aas.impl.metamodel.hashmap.aas.qualifier.haskind;

import java.util.HashMap;

import org.eclipse.basyx.aas.api.metamodel.aas.qualifier.haskind.IHasKind;
import org.eclipse.basyx.aas.impl.metamodel.facades.HasKindFacade;

/**
 * HasKind class
 * 
 * @author elsheikh, schnicke
 *
 */
public class HasKind extends HashMap<String, Object> implements IHasKind {

	/**
	 * Version of serialized instances
	 */
	private static final long serialVersionUID = 1L;
	
	public static final String KIND="kind";

	/**
	 * Constructor
	 */
	public HasKind() {
		// Default value

		put(KIND, null);
	}

	/**
	 * Constructor that takes
	 * {@link org.eclipse.basyx.aas.impl.metamodel.hashmap.aas.qualifier.haskind.Kind
	 * Kind}(either Kind.Instance or Kind.Type)
	 */
	public HasKind(String kind) {
		// Kind of the element: either type or instance.

		put(KIND, kind);
	}

	@Override
	public String getHasKindReference() {
      return new HasKindFacade(this).getHasKindReference();
	}

	@Override
	public void setHasKindReference(String kind) {
		new HasKindFacade(this).setHasKindReference(kind);
		
	}
}

File: sdks/c++/basys.sdk.cc/src/submodel/submodel/map/qualifier/haskind/HasKind.cpp
/*
 * HasKind.cpp
 *
 *      Author: wendel
 */

#include "HasKind.h"

namespace basyx {
namespace submodel {
namespace metamodel {
namespace map {
namespace qualifier {
namespace haskind {

HasKind::HasKind() :
  kind(Kind::NOTSPECIFIED)
{}

HasKind::HasKind(Kind kind) :
  kind(kind)
{}

qualifier::haskind::Kind HasKind::getHasKindReference() const
{
  return this->kind;
}

void HasKind::setHasKindReference(Kind kind)
{
  this->kind = kind;
}

}
}
}
}
}
}